transportable-udfs-type-system/src/test/java/com/linkedin/transport/typesystem/AbstractTestStdFactoryForTypes.java

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright 2023 LinkedIn Corporation. All rights reserved.
+ * Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.transport.typesystem;
+
+import com.linkedin.transport.api.StdFactory;
+import com.linkedin.transport.api.types.StdStructType;
+import com.linkedin.transport.api.types.StdType;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.assertEquals;
+
+
+public abstract class AbstractTestStdFactoryForTypes {
```

Comment:
Few thoughts:
1. Just wonder if we need to create this new test for testing parsing signature to StdType? It looks like an integration test to me and each individual conversion should be covered in separate unit tests.

If we dont need such integration test, then @phd3 you can just add a few lines to check for the new method you introduced for StdStructType in each type system. For example you can add some test code for that method in [Avro here](https://github.com/linkedin/transport/blob/09a89508296a2491f43cc8866d47952c911313ab/transportable-udfs-avro/src/test/java/com/linkedin/transport/avro/TestAvroWrapper.java) or [Spark here](https://github.com/linkedin/transport/blob/master/transportable-udfs-spark_2.11/src/test/scala/com/linkedin/transport/spark/TestSparkFactory.scala). Though I could not find similar tests for Hive and Trino and seems we missed that @wmoustafa?

In this way, @phd3 you dont have to copy the `TrinoTestFunctionDependencies` code here, avoid duplicate code issue?

2. If we still want to do such integration test here, then a minor to rename the test: TestSignatureToStdType for better meaning.

Reply:
Yeah - seems okay to just add individual unit tests.